pkgs/development/libraries/vapoursynth/default.nix

```diff
@@ -42,6 +47,60 @@ stdenv.mkDerivation rec {
     # wrapper, what python3 version was used to build vapoursynth so
     # the right python3.sitePackages will be used there.
     inherit python3;
+
+    withPlugins = plugins: let
+      pythonEnvironment = python3.buildEnv.override {
+        extraLibs = plugins;
+      };
+
+      getRecursivePropagatedBuildInputs = pkgs: lib.flatten
+        (map
+          (pkg: pkg.propagatedBuildInputs ++ (getRecursivePropagatedBuildInputs pkg.propagatedBuildInputs))
+          pkgs);
+
+      deepPlugins = plugins ++ (getRecursivePropagatedBuildInputs plugins);
+
+      pluginLoader = let
+        source = writeText "vapoursynth-nix-plugins.c" ''
+          void VSLoadPluginsNix(void (*load)(void *data, const char *path), void *data) {
+          ${concatMapStringsSep "" (path: "load(data, \"${path}/lib/vapoursynth\");") deepPlugins}
+          }
+        '';
+      in
+      runCommandCC "vapoursynth-plugin-loader" {
+        executable = true;
+        passAsFile = ["code"];
+        preferLocalBuild = true;
+        allowSubstitutes = false;
+      } ''
+        mkdir -p $out/lib
+        $CC -shared -fPIC ${source} -o "$out/lib/libvapoursynth-nix-plugins${ext}"
+      '';
+
+      ext = stdenv.targetPlatform.extensions.sharedLibrary;
+    in
+    buildEnv {
+      name = "${vapoursynth.name}-with-plugins";
+      paths = [ vapoursynth pluginLoader pythonEnvironment ];
+      buildInputs = [ makeWrapper patchelf ];
+      postBuild = ''
+        rm $out/lib/libvapoursynth${ext}
+        cp ${vapoursynth}/lib/libvapoursynth${ext} $out/lib/libvapoursynth${ext}
+        chmod +w $out/lib/libvapoursynth${ext}
+        patchelf $out/lib/libvapoursynth${ext} \
+            --add-needed libvapoursynth-nix-plugins${ext}
+        chmod -w $out/lib/libvapoursynth${ext}
+
+        rm -r $out/bin
+        makeWrapper ${vapoursynth}/bin/vspipe $out/bin/vspipe \
+            --prefix LD_LIBRARY_PATH : $out/lib \
+            --prefix PYTHONPATH : $out/${python3.sitePackages}
+      '';
```

Comment:
```suggestion
    runCommand "${vapoursynth.name}-with-plugins" {
      nativeBuildInputs = [ makeWrapper ];
    } ''
      mkdir -p \
        $out/bin \
        $out/lib/pkgconfig \
        $out/lib/vapoursynth \
        $out/${python3.sitePackages}

      for textFile in \
          lib/pkgconfig/vapoursynth{,-script}.pc \
          lib/libvapoursynth.la \
          lib/libvapoursynth-script.la \
          ${python3.sitePackages}/vapoursynth.la
      do
          substitute ${vapoursynth}/$textFile $out/$textFile \
              --replace "${vapoursynth}" "$out"
      done

      for binaryFile in \
          lib/libvapoursynth${ext} \
          lib/libvapoursynth-script${ext}.0.0.0
      do
        old_rpath=$(patchelf --print-rpath ${vapoursynth}/$binaryFile)
        new_rpath="$old_rpath:$out/lib"
        patchelf \
            --set-rpath "$new_rpath" \
            --output $out/$binaryFile \
            ${vapoursynth}/$binaryFile
        patchelf \
            --add-needed libvapoursynth-nix-plugins${ext} \
            $out/$binaryFile
      done

      for binaryFile in \
          ${python3.sitePackages}/vapoursynth${ext} \
          bin/.vspipe-wrapped
      do
          old_rpath=$(patchelf --print-rpath ${vapoursynth}/$binaryFile)
          new_rpath="''${old_rpath//"${vapoursynth}"/"$out"}"
          patchelf \
              --set-rpath "$new_rpath" \
              --output $out/$binaryFile \
              ${vapoursynth}/$binaryFile
      done

      ln -s \
          ${pluginLoader}/lib/libvapoursynth-nix-plugins${ext} \
          $out/lib/libvapoursynth-nix-plugins${ext}
      ln -s ${vapoursynth}/include $out/include
      ln -s ${vapoursynth}/lib/vapoursynth/* $out/lib/vapoursynth
      ln -s \
          libvapoursynth-script${ext}.0.0.0 \
          $out/lib/libvapoursynth-script${ext}
      ln -s \
          libvapoursynth-script${ext}.0.0.0 \
          $out/lib/libvapoursynth-script${ext}.0

      makeWrapper $out/bin/.vspipe-wrapped $out/bin/vspipe \
          --prefix PYTHONPATH : $out/${python3.sitePackages}
    '';
```

I think having to wrap vspipe shows that this solution was actually incomplete. Any other binary depending on libvapoursynth-script would have to be wrapped in this way, but this change shouldn't require changes in other packages.

I think this new build gets replaces all the vapoursynth references now.

Reply:
Thanks! I applied the changes and re-added support for python plugins.